src/streams.h

```diff
@@ -718,7 +718,7 @@ class CBufferedFile
     uint64_t nReadPos;    //!< how many bytes have been read from this
     uint64_t nReadLimit;  //!< up to which position we're allowed to read
     uint64_t nRewind;     //!< how many bytes we guarantee to rewind
-    std::vector<char> vchBuf; //!< the buffer
+    std::vector<unsigned char> vchBuf; //!< the buffer
 
 protected:
```

Comment:
While the `CBufferedFile` class is already touched, maybe make `CBufferedFile::Fill` private by removing this line?